### PR TITLE
[WB-9795] Launch agent config sets namespace

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-9795](https://wandb.atlassian.net/browse/WB-9795)
Fixes #NNNN

Description
-----------
If the user sets the kubernetes namespace in the launch agent config file, kubernetes will always use that namespace

Testing
-------
How was this PR tested?

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/sdk/launch/runner/kubernetes.py [hunk 1]
```diff
@@ -233,6 +233,14 @@ def wait_job_launch(
         )
         return pod_names
 
+    def get_namespace(
+        self, resource_args: Dict[str, Any], default_namespace: str
+    ) -> str:
+        runner_config_namespace = self.backend_config.get("runner", {}).get("namespace")
+        if runner_config_namespace is not None:
+            return runner_config_namespace  # type: ignore
+        return resource_args.get("namespace", default_namespace)
```
Comment (hu-po) [inline wandb/sdk/launch/runner/kubernetes.py:242]: maybe just make this the default in the `.get("namespace")` line?

**Review by hu-po — APPROVED** [review]
I wonder if there is a better pattern for this type of default-resolution-order problem. Seems like this is ripe for some weird edge cases.

**Comment by KyleGoyette** [comment]
Yeah, we could potentially have a single place where we merge a backend config and override configs... the only issue is this is a weird case where we want default to override any passed in value...

I'll file a ticket to make a single function to resolve a config between backend and passed in config values.

File: wandb/sdk/launch/runner/kubernetes.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/runner/kubernetes.py:242]: Nice

**Pull request merged**